### PR TITLE
Fuzzing v128 and associated bug fixes

### PR DESCRIPTION
**Pull request by tlively**

### CODE REVIEW

File: src/tools/fuzzing.h [hunk 1]
```diff
@@ -177,6 +177,7 @@ class TranslateToFuzzReader {
 
   // Optionally remove NaNs, which are a source of nondeterminism (which makes
   // cross-VM comparisons harder)
+  // TODO: de-NaN SIMD values
```
Comment (tlively) [inline src/tools/fuzzing.h]: @kripken how critical is this given that the default setting is to enable SIMD values in the fuzzer?

File: src/wasm/wasm.cpp [hunk 2]
```diff
@@ -381,13 +381,15 @@ void SetGlobal::finalize() {
 }
 
 void Load::finalize() {
+  assert(!isAtomic || type == i32 || type == i64 || type == unreachable);
```
Comment (kripken) [inline src/wasm/wasm.cpp]: I think this and the assert below should be covered by the validator (not in an assert, but also in release builds there).

File: src/tools/fuzzing.h (hunk 1 above)
Comment (kripken) [reply src/tools/fuzzing.h]: Well, NaNs can cause some nondeterminism, which makes comparing between VMs hard. But definitely something we can leave for later.

File: src/wasm/wasm.cpp (hunk 2 above)
Comment (tlively) [reply src/wasm/wasm.cpp]: Done.

**Comment by tlively** [comment]
I'll handle the `idx` -> `index` renaming in another PR. Thanks for catching that.

**Pull request merged**